tablo/models.py

```diff
@@ -995,30 +1013,41 @@ def create_aggregate_database_table(row, dataset_id):
     data_types = ['String']
     optional_fields = []
     type_conversion = {
-        'decimal': 'Decimal',
+        'double': 'Decimal',
         'date': 'Date',
         'integer': 'Integer',
         'string': 'String',
         'xLocation': 'Decimal',
         'yLocation': 'Decimal',
-        'dropdownedit': 'String',
+        'dropdownEdit': 'String',
         'dropdown': 'String',
     }
+    date_fields = []
     for col in row:
         col_name = convert_header_to_column_name(col.column)
         columns.append(col_name)
         data_types.append(type_conversion[col.type])
+        if col.type == 'date':
+            date_fields.append(col_name)
         if hasattr(col, 'required') and not col.required:
             optional_fields.append(col_name)
     df = pd.DataFrame(columns=columns)
+    for field in date_fields:
+        df[field] = df[field].astype('datetime64')
     df_info = {
         'dataTypes': data_types,
         'optionalFields': optional_fields
     }
-    table_name = create_database_table(df, df_info, dataset_id)
+    table_name = create_database_table(df, df_info, dataset_id, append=True)
```

Comment:
This happens in the context of the `create_aggregate_database_table` method: why do we need the `append` argument?

I'm asking because, if it causes issues here for it to be `False`, when wouldn't it cause issues?

Reply:
It's because of the way the aggregate pipeline works: it creates the temp import table twice, which causes some issues with sequences. I didn't have time to look more into this to see why it's behaving the way it does, but considering the tables come in empty, adding `append=True` fixes the issue without any side effect. I also modified `create_database_table` to handle empty tables.

Later we probably need to overhaul the import and aggregate process, so we can optimize this then. I can add a `TODO` tag here for future reference.